From 08968eb2bfd306b4150e8531f799adee6a11966c Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Mar 2017 13:25:13 +0100
Subject: [PATCH] tc: flower: Fix parsing ip address

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1422629
Upstream Status: iproute2.git commit 164a9ff401e58

commit 164a9ff401e58ff9c0b90088a3aa96216e1d6e8a
Author: Roi Dayan <roid@mellanox.com>
Date:   Wed Feb 22 16:05:01 2017 +0200

    tc: flower: Fix parsing ip address

    Fix order of arguments when passed to __flower_parse_ip_addr.

    Fixes: ("f888f4e20534 tc: flower: Support matching ARP")
    Signed-off-by: Roi Dayan <roid@mellanox.com>
    Reviewed-by: Paul Blakey <paulb@mellanox.com>
    Reviewed-by: Simon Horman <simon.horman@netronome.com>
---
 tc/f_flower.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tc/f_flower.c b/tc/f_flower.c
index e9d3a96..17e9b3a 100644
--- a/tc/f_flower.c
+++ b/tc/f_flower.c
@@ -289,8 +289,8 @@ static int flower_parse_ip_addr(char *str, __be16 eth_type,
 		return -1;
 	}
 
-	return __flower_parse_ip_addr(str, family, addr4_type, addr6_type,
-				      mask4_type, mask6_type, n);
+	return __flower_parse_ip_addr(str, family, addr4_type, mask4_type,
+				      addr6_type, mask6_type, n);
 }
 
 static bool flower_eth_type_arp(__be16 eth_type)
-- 
1.8.3.1

